streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -66,10 +67,11 @@ public void run() {
     };
 
     private long timestamp;
+    private Punctuator punctuator;
```

Comment:
I think they should be local to the `punctuate` method as they are only needed there. If it becomes a problem they can be optimized later. I don't see the point in optimizing before there is a problem, especially when it is making the code worse (IMO)